portfolio/src/main/webapp/script.js

```diff
@@ -153,8 +156,23 @@ function formatCommentText(comment) {
   return quote;
 }
 
-// Sets value of form submission time to current time in client's timezone
-function inputClientTime(form) {
+/**
+ * Sets value of form submission time to current time in client's timezone,
+ * posts form data to server, reloads comments section and then clears
+ * form in preparation for next entry
+ */
+function submitForm(form) {
   const today = new Date();
-  form.time.value = today.toLocaleString();
+  form.timestamp.value = today.getTime();
+  const formData = JSON.stringify($("#newcommentform").serialize());
```

Comment:
I think jquery serialize creates this `key=value&key=value` format. Instead you could use `serializeArray` and create a JSON object out of it:

```
const formData = {};
$("#newcommentform").serializeArray().forEach(entry => {
  formData[entry.name] = entry.value;
});
```

and send `JSON.stringify(formData)` in the request.

This way you don't need your own string parsing in Java and can use Gson JsonParser.